client/verta/tests/unit_tests/test_endpoint.py

```diff
@@ -69,212 +69,238 @@ def mock_endpoint(mock_conn, mock_config) -> Endpoint:
-            mock_registered_model_version, kafka_settings=kafka_settings
+    GET_JSON_BY_ID_RESPONSE_MISSING_URL.pop(
+        "full_url"
+    )  # Remove the full url key and value from the dict.
+
+    @patch(f"{VERTA_CLASS}.get_status", return_value=GET_STATUS_RESPONSE)
+    @patch(
+        f"{VERTA_CLASS}._get_json_by_id",
+        return_value=GET_JSON_BY_ID_RESPONSE_MISSING_URL,
+    )
+    @patch(f"{VERTA_CLASS}.get_access_token", return_value=GET_ACCESS_TOKEN_RESPONSE)
+    def test_get_deployed_model_missing_full_url(
+        self,
+        mock_get_access_token,  # pass in patched methods in order
+        mock_get_json_by_id,
+        mock_get_status,
+        mock_endpoint,
+    ) -> None:
+        """Verify that the get_deployed_model method returns the correct value for the full_url when it is returned
+        by the get_json_by_id method.  The 'full_url' key was added to the schema of the response 10/2022.
+        """
+        deployed_model = mock_endpoint.get_deployed_model()
+        assert (
+            deployed_model.prediction_url
+            == "https://test_socket/api/v1/predict/test_path"
         )
 
 
-@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
-@given(
-    kafka_settings=mock_kafka_settings(),
-    kafka_configs_response=mock_kafka_configs_response(),
-)
-def test_kafka_config_missing_config_id_exception(
-    kafka_settings,
-    kafka_configs_response,
-    mock_endpoint,
-    mock_conn,
-    mocked_responses,
-    mock_registered_model_version,
-) -> None:
-    """In the unlikely evert the ID of a found Kafka config is missing from the
-    backend response, the expected exception is raised.
-    """
-    kafka_configs_response["configurations"][0].pop("id")
-    deployment_url = f"{mock_conn.scheme}://{mock_conn.socket}/api/v1/deployment"
-    stages_url = (
-        f"{deployment_url}/workspace/{WORKSPACE_ID}/endpoints/{DEPLOYMENT_ID}/stages"
+class TestKafka:
+    @settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+    @given(
+        kafka_settings=mock_kafka_settings(),
+        kafka_configs_response=mock_kafka_configs_response(),
     )
-    get_configs_url = f"{mock_conn.scheme}://{mock_conn.socket}/api/v1/uac-proxy/system_admin/listKafkaConfiguration"
+    def test_kafka_cluster_config_id_default(
+        self,
+        kafka_settings,
+        kafka_configs_response,
+        mock_endpoint,
+        mock_conn,
+        mocked_responses,
+        mock_registered_model_version,
+    ) -> None:
+        """Verify that, while updating an endpoint, not including a `cluster_config_id`
+        in the KafkaSettings results in the correct sequence of HTTP requests, including
+        a call to fetch the ID of the current Kafka config by default."""
+        deployment_url = f"{mock_conn.scheme}://{mock_conn.socket}/api/v1/deployment"
+        stages_url = f"{deployment_url}/workspace/{WORKSPACE_ID}/endpoints/{DEPLOYMENT_ID}/stages"
```

Comment:
This syntax is something `black` did when run on this file.  In fact, if I take your version of the file and run black against it, several of your edits are undone.  Perhaps we should make sure we are in sync.